week6/src/CreateTask.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState,useEffect } from 'react';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import { Link} from 'react-router-dom';
+import List from './List';
+
+
+function CreateTask({modal,toggle}) {
+    const linkStyle = {
+        textDecoration: "none",
+        color: 'black'
+      };
+  const storedTask = JSON.parse(localStorage.getItem('items')) || []
+  const storedDescription = JSON.parse(localStorage.getItem('narration')) || []
+  const storedDate = JSON.parse(localStorage.getItem('date')) || []
+  const [tasks, setNewitem]= useState("");
+  const [items, setItems] = useState(storedTask);
+  const [descriptions, setNewDescription]= useState("");
+  const [narration, setDescription] = useState(storedDescription);
+  const [time, setNewDate]= useState("");
+  const [date, setDate] = useState(storedDate);
+  useEffect(() => {
+    localStorage.setItem('items', JSON.stringify(items));
+    
+  }, [items]);
+  useEffect(() => {
+    localStorage.setItem('narration', JSON.stringify(narration));
+    
+  }, [narration]);
+  useEffect(() => {
+    localStorage.setItem('date', JSON.stringify(date));
+    
+  }, [date]);
+  function add(){
+    if (!tasks){
+      alert("please input task!");
+      return;
+    }
+    if (!descriptions){
+        alert("please input description!");
+        return;
+      }
+      if (!time){
+        alert("please input date!");
+        return;
+      }
+    const taskinput = /^(?=.*[a-z])/;
+    if (!taskinput.test(tasks)){
+      alert('task must contain at least one letter in');
+      return;
+    }
+    const descriptionInput = /^(?=.*[a-z])/;
+      if (!descriptionInput.test(descriptions)){
+        alert('description must contain at least one letter');
+        return;
+      }
+      const dateInput = /^(?=.*[0-9])/;
+      if (!dateInput.test(time)){
+        alert('date must contain at least one digit');
+        return;
+      }
+
+    const item = {
+      id: Math.floor(Math.random() * 1000),
+      value: tasks
+    };
+    
+    setItems(oldList => [...oldList, item]);
+    setNewitem(""); 
+  
+      const description = {
+        id: item.id,
+        value: descriptions
+      };
+      
+      setDescription(oldList => [...oldList, description]);
+      setNewDescription("");
+
+      const period = {
+        id: item.id,
+        value: time
+      };
+      
+      setDate(oldList => [...oldList, period]);
+      setNewDate("");
+      toggle();
+   }
+   function Delete(id){
+    const newArray1 = items.filter(item => item.id !== id);
+    const newArray2 = narration.filter(description => description.id !== id);
+    const newArray3 = date.filter(date => date.id !== id);
+    setItems(newArray1);
+    setDescription(newArray2);
+    setDate(newArray3);
+     } 
+  return (
+    <div>
+      <Modal isOpen={modal} toggle={toggle}>
```

Comment:
you can see that 
1. the component is called Modal
2. you are passing a prop value called modal too. Could we name is isOpen instead? Apply this same process to the other places we use modal wrongly